pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
```

Comment:
Maybe high scale services should have an alternate endpoint that handles that scale more effectively, where you are *not* allowed to get all endpoints from the endpoints/ call, and there's a "get more" link out to that alternate form.  I assume the high scale consumers of headless servers know they are high scale, so they can transparently switch from one mode to another.

----- Original Message -----
> > @@ -738,7 +771,35 @@ type Endpoints struct {
> >  	TypeMeta   `json:",inline"`
> >  	ObjectMeta `json:"metadata,omitempty"`
> >  
> > -	Endpoints []string `json:"endpoints,omitempty"`
> > +	Endpoints []Endpoint `json:"endpoints,omitempty"`
> 
> @thockin SGTM.
> 
> @smarterclayton Mostly for headless services, such as petasort mapreduce.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4370/files#r24844801